tests/unit/historyController/Test.js

```diff
@@ -123,27 +119,26 @@ define([
 					}
 				}
 			};
-			testApp.showOrHideViews('hc1center3', params);
-			return displayDeferred.then(function () {
-				var hc1center3content = document.getElementById("hc1center3");
-				var hc1center3View = viewUtils.getViewFromViewId(testApp, "hc1center3");
-				checkNodeVisibile(hc1center3content);
-				checkActivateCallCount(hc1center3View, 1, true);
-			}).then(function () {
-				var currentHash = window.location.hash;
-				assert.strictEqual(currentHash,
-					"#hc1header1+hc1centerParent+(hc1center3&pTestVal1=value1)+hc1right1+hc1footer1",
-					" currentHash should include (hc1center3&pTestVal1=value1)+hc1right1+hc1footer1");
-			});
+			return when(testApp.showOrHideViews('hc1center3', params)
+				.then(function () {
+					var hc1center3content = document.getElementById("hc1center3");
+					var hc1center3View = viewUtils.getViewFromViewId(testApp, "hc1center3");
+					checkNodeVisibile(hc1center3content);
+					checkActivateCallCount(hc1center3View, 1, true);
+					var currentHash = window.location.hash;
+					assert.strictEqual(currentHash,
+						"#hc1header1+hc1centerParent+(hc1center3&pTestVal1=value1)+hc1right1+hc1footer1",
+						" currentHash should include (hc1center3&pTestVal1=value1)+hc1right1+hc1footer1");
+				}));
 		},
 		// Currently showing hc1header1+hc1centerParent+hc1center3+hc1right1+hc1footer1 test
 		// history.back()
 		"test history.back() to get back to hc1center2)": function () {
 			this.timeout = 20000;
-			var displayDeferred = new Deferred();
-			setupOnOnce(testApp, displayDeferred);
-			history.back();
-			return displayDeferred.then(function () {
+			return when(new Promise(function (resolve) {
+				setupOnOncePromise(testApp, resolve);
+				history.back();
+			}).then(function () {
```

Comment:
I thought you were going to make setupOnOncePromise() return a promise?   Although it turn out that you have to pass in function () { history.back(); }.